lib/spack/spack/architecture.py

```diff
@@ -498,6 +498,16 @@ def platform():
             return platform_cls()
 
 
+@memoized
+def front_end_sys_type():
```

Comment:
if we're writing it "frontend" everywhere else, this should be `frontend_sys_type`